proposals/0271-swiftpm-binary-dependencies.md

```diff
@@ -0,0 +1,417 @@
+a built-in and automatic feature of the package manager which applies to all
+packages.
+
+### Binary Signatures
+
+We considered adding signature checks during the checkout of binary dependencies but when these have transitive dependencies it gets complicated expressing that in the `Package.swift`.
+
+```
+     let package = Package(
+         name: "Paper",
+         products: [...],
+         dependencies: [
+             .package(url: "http://some/other/lib", .exact("1.2.3"), binarySignature: .any),
+             .package(url: "http://some/other/lib", .exact("1.2.3"), binarySignature: .gpg("XXX")"),
+         ],
+         targets: [...]
+     )
+```
+
+### Binary target vs. binary product
+During the discussion, it was brought up wether a binary dependency should be declared as a target or a product. Below is a list of what we took into consideration when deciding between target and product. In the end, a target seems like a better choice for a binary dependency.
+
+- Targets allow configuring linker/compiler flags; this ability might be necessary for static libraries
+- There are already `systemLibraryTargets` which are essentially dependencies on pre-existing binaries on the host system
+- Targets represent a single module, the same is true for one XCFramework
+- Currently there is no way to depend on products, so mixed binary and source packages might be harder if we go with a binary product approach
+- Analogy with what currently is being produced by source packages (dylibs => product)
+
+### .o file format
+During the discussion of the proposal, the idea of using `.o` files was brought up. This would follow what SwiftPM creates for source-based dependencies right now; therefor, making the integration potentially easier. The main benefit of using `.o` files would be that the product linking the binary artifact can decide wether to link it dynamically or statically. However, further discussion needs to happen here how this would work. For now XCFrameworks are the initial format. XCFrameworks will allow current framework authors to package their existing XCFrameworks and distribute them via SwiftPM.
+
+### Avoiding duplicate symbols with static libraries/frameworks
+When multiple products depend on a static library or framework it will result in duplicated symbols. SwiftPM could be smart enough to figure this out from the manifest and provide an error. This is a potential future improvement, but would require SwiftPM to know the linkage type of the artifact.
+
+### Opt-in to allow binaries
+In the beginning, we considered to make binary support a opt-in feature so that only when somebody explicitly allows it then SwiftPM tried to use them. However, after discussion in the Swift forum we came to the conclusion that the trust one has to extend to a dependency is no different between a source-based and binary-based one; therefor, we made removed the opt-in behavior but added a opt-out behavior.
```

Comment:
```suggestion
In the beginning, we considered making binary support an opt-in feature so that only when somebody explicitly allows it then SwiftPM tried to use them. However, after discussion in the Swift forum we came to the conclusion that the trust one has to extend to a dependency is no different between a source-based and binary-based one; therefore, we removed the opt-in behavior but added an opt-out behavior.
```